topics/data-science/tutorials/python-loops/tutorial.md

```diff
@@ -0,0 +1,676 @@
+for char in original:
+    result = ____
+print(result)
+```
+
+
+> ### {% icon question %} Question: Practice Accumulating
+>
+> Fill in the blanks in each of the programs below
+> to produce the indicated result.
+>
+> ```
+> # Total length of the strings in the list: ["red", "green", "blue"] => 12
+> total = 0
+> for word in ["red", "green", "blue"]:
+>     ____ = ____ + len(word)
+> print(total)
+> ```
+>
+> > ### {% icon solution %} Solution
+> > ```
+> > total = 0
+> > for word in ["red", "green", "blue"]:
+> >     total = total + len(word)
+> > print(total)
+> > ```
+> {: .solution}
+>
+> ```
+> # List of word lengths: ["red", "green", "blue"] => [3, 5, 4]
+> lengths = ____
+> for word in ["red", "green", "blue"]:
+>     lengths.____(____)
+> print(lengths)
+> ```
+> > ### {% icon solution %} Solution
+> > ```
+> > lengths = []
+> > for word in ["red", "green", "blue"]:
+> >     lengths.append(len(word))
+> > print(lengths)
+> > ```
+> {: .solution}
+>
+> ```
+> # Concatenate all words: ["red", "green", "blue"] => "redgreenblue"
+> words = ["red", "green", "blue"]
+> result = ____
+> for ____ in ____:
+>     ____
+> print(result)
+> ```
+> > ### {% icon solution %} Solution
+> > ```
+> > words = ["red", "green", "blue"]
+> > result = ""
+> > for word in words:
+> >     result = result + word
+> > print(result)
+> > ```
+> {: .solution}
+>
+> __Create an acronym:__ Starting from the list `["red", "green", "blue"]`, create the acronym `"RGB"` using
+> a for loop.
+>
+> __Hint:__ You may need to use a string method to properly format the acronym.
+> > ### {% icon solution %} Solution
+> > ```
+> > acronym = ""
+> > for word in ["red", "green", "blue"]:
+> >     acronym = acronym + word[0].upper()
+> > print(acronym)
+> > ```
+> {: .solution}
+{: .question}
+
+```python
+#Test your code here!
+```
+
+> ## Cumulative Sum
+>
+> Reorder and properly indent the lines of code below
+> so that they print a list with the cumulative sum of data.
+> The result should be `[1, 3, 5, 10]`.
+>
+> ```
+> cumulative.append(total)
+> for number in data:
+> cumulative = []
+> total += number
+> total = 0
+> print(cumulative)
+> data = [1,2,2,5]
+> ```
+> > ### {% icon solution %} Solution
+> > ```
+> > total = 0
+> > data = [1,2,2,5]
+> > cumulative = []
+> > for number in data:
+> >     total += number
+> >     cumulative.append(total)
+> > print(cumulative)
+> > ```
+> {: .solution}
+{: .question}
+
+```python
+# Test your code here!
+```
+
+
+> ### {% icon question %} Question: Identifying Variable Name Errors
+>
+> 1. Read the code below and try to identify what the errors are
+>    **without** running it.
+> 2. Run the code and read the error message.
+>    What type of `NameError` do you think this is?
+>    Is it a string with no quotes, a misspelled variable, or a
+>    variable that should have been defined but was not?
+> 3. Fix the error.
+> 4. Repeat steps 2 and 3, until you have fixed all the errors.
+>
+> ```
+> for number in range(10):
```

Comment:
this really reminds me of the bee boop small programming test>
Maybe add is as an excersize?